bundles/org.openhab.core.io.rest.auth/src/main/java/org/openhab/core/io/rest/auth/internal/AuthFilter.java

```diff
@@ -267,4 +276,55 @@ public void filter(@Nullable ContainerRequestContext requestContext) throws IOEx
             }
         }
     }
+
+    private boolean isImplicitUserRole(ContainerRequestContext requestContext) {
+        if (implicitUserRole) {
+            return true;
+        }
+        if (trustedNetworks.isEmpty()) {
+            return false;
+        }
```

Comment:
This is not needed, if `trustedNetworks` is an empty list, `Stream.anyMatch` also returns false.